Assets/SEE/Game/Worlds/PlayerSpawner.cs

```diff
@@ -54,16 +63,24 @@ private void OnEnable()
         /// <returns>enumerator as to how to continue this co-routine</returns>
         private IEnumerator SpawnPlayerCoroutine()
         {
-            NetworkManager networkManager = NetworkManager.Singleton;
+            networkConfig = FindObjectOfType<SEE.Net.Network>();
+            if (networkConfig == null)
+            {
+                Debug.LogError("Network configuration not found");
+                yield return null;
```

Comment:
I am not familiar with yield and coroutines. But i now think this is totally wrong. :) Shouldn't it be like this (similiar to the networkManager check):
```
while (ReferenceEquals(networkConfig, null))
{
    Debug.LogError("Network configuration not found");
    networkConfig = FindObjectOfType<SEE.Net.Network>();
    yield return null;
}
```
My understanding would be that routine pause and try to find the missing networkConfig later again?

Reply:
Yes, something like this does look a lot more reasonable. I'd even recommend using `async` functionality here instead:
```csharp
await UniTask.WaitUntil(() => !ReferenceEquals(networkConfig = FindObjectOfType<SEE.Net.Network>(), null));
```
This should have the same effect as the coroutine approach above (checking every frame whether the Network object is present). I don't know if this is really necessary at all, though—if we do keep it, we should remove the error message (like in the `async` example above), otherwise the message will appear every frame.

Reply:
This is something i am still struggeling with. To use your code snippet i need to make the whole Coroutine a UniTask and remove everything which is like "yield return null" as i understand. But i didn't get it running without freezing :/

Reply:
Hmm, okay. Just to check, does the following work? (Untested, just to give you an idea of what it would look like to make this method fully async)

## Method itself
```csharp
networkConfig = FindObjectOfType<SEE.Net.Network>();
await UniTask.WaitUntil(() => !ReferenceEquals(networkConfig = FindObjectOfType<SEE.Net.Network>(), null));
await UniTask.WaitUntil(() => !ReferenceEquals(networkManager = NetworkManager.Singleton, null));

// Terminate this function if not run by the server, but first set client playername for chat-function.
if (!networkManager.IsServer)
{
    dissonanceComms = FindObjectOfType<DissonanceComms>();
    dissonanceComms.LocalPlayerName = networkConfig.PlayerName;
    return;
}

// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// The following code will be executed only on the server.
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// Wait until Dissonance is created
await UniTask.WaitUntil(() => !ReferenceEquals(dissonanceComms = FindObjectOfType<DissonanceComms>(), null));
dissonanceComms.LocalPlayerName = networkConfig.PlayerName;

// The callback to invoke once a client connects. This callback is only
// ran on the server and on the local client that connects. We want
// to spawn a player whenever a client connects.
networkManager.OnClientConnectedCallback += Spawn;
networkManager.OnClientDisconnectCallback += ClientDisconnects;
// Spawn the local player. This code is executed by the server.
Spawn(networkManager.LocalClientId);
```

## Method signature
```csharp
private async UniTaskVoid SpawnPlayer()
```

## Method usage
AFAICT, this method is only used in `OnEnable`, but any other usages would need to be replaced as well:
```csharp
// Previously: StartCoroutine(SpawnPlayerCoroutine());
SpawnPlayerCoroutine().Forget();
```

---

If this still doesn't work (e.g., freezes), we could open a follow-up issue for this instead of doing it in this PR if that takes too much time. There are some other places in SEE that need to be updated to be async anyway.

Reply:
Looks similar to one of my attempts. Tried this again but not working. :( It freezes and can't set up playername, it says dissonance is started already and then the playername can't be set.

Reply:
Oh well, not sure what causes this—I created a follow-up issue for this now (#762) as it's not essential for this PR, so we can close this thread.